tests/test_flask.py

```diff
@@ -712,6 +712,10 @@ def test_workout(client):
     assert returned_value.status_code == 200
 
     # TODO: need to test post requests
+    g.database.add_workout("test","test","test","test","test")
+    assert g._init_.workout("test") != g._init_.abort(404)
+
+
```

Comment:
GH is not allowing me to select the whole test for some reason so treat this as if I highlighted the whole test. 

- Rewrite the test (remove all my old code since it's placeholder) and first do a `response_value = client.get()` to get the test workout created in the setUp function in the beginning of the file. This should redirect the user to the login page, so `assert response_value.status_code == 200`. Then `assert b'login' in response_value.data` to check if its the login page. 
- Then login as testTrainee by doing a `login_as_testTrainee()`,  and every request to `client` will be done as the testTrainee user. 
  - Try to add a test workout. 
  - Check if testTrainee can see the workout
  - Delete the workout for clean up
- Then login as testTrainer by doing a `login_as_testTrianee()` and repeat the above. 
- Then check if a non-existent workout returns a 404.